### DISCUSSION TITLE
Enrichment: How to get a FWD Node grab the hostname of the Source and/or destination IP for internal logs?

### DISCUSSION THREAD

**Discussion by wildlyunder** [Security-Onion-Solutions/securityonion#6849] [Unsupported Versions] [answered]
**Background**
We have a number of FWD nodes in a number of security zones, each with there own DNS and DHCP servers and all airgapped for each other. Our SO Manager is also in it's own security zone and unable to query anything beyond it's security zone and can only receive logs from the FWD Nodes in each zone.

Currently our analyst team is having to manually resolve IP's to Hostnames which is taking up a huge amount of effort when we are performing IR or Hunting due to the large number of Security Zone we have. I.E They need to jump into each zone and do a manual lookup for each IP if they need to resolve the hostname. This isn't ideal if the IP is dynamic. 

**Question**:
When a logs is sent to a FWD node, how can we enrich the logs with the hostname of the source and destination IP before it is sent onto the SIEM.

@dougburks Do you know if this is possible?

**Comment by dougburks** [comment, marked as answer]
> @dougburks Do you know if this is possible?

Please do not tag an individual in a discussion unless they have already volunteered to help you in that discussion.

You might be able to configure Filebeat on the forward node to resolve IP addresses to hostnames:
https://www.elastic.co/guide/en/beats/filebeat/current/processor-dns.html

However, please note the warnings about DNS lookups affecting pipeline performance.

**Comment by wildlyunder** [comment]
HI Doug,

Sorry for that, I saw other people tag people in. I thought that was the allowed in this forum. I won't do it in future.
